### PR TITLE
OBS tweaks

### PR DESCRIPTION
**Pull request by WillB97**
- adds a global option for pre-play transition time: `preload_time`
- splits load-transition and play into separate scheduler events
- changes obs-video action start time to point at when video playback commences
- updates `example_playlist.yaml` to include OBS config

### CODE REVIEW

**Comment by PeterJCLaw** [comment]
Thanks for putting this together. Do you think we could break the addition of the transition-only action into its own PR? That would just leave here the adjustments around how the preload/trasnition/play split is handled, which I think is more likely to attract a variety of opinions.

**Review by RealOrangeOne — APPROVED** [review]
This code looks probably fine, but i've not tested it.

File: example-playlist.yaml [hunk 1]
```diff
@@ -1,3 +1,9 @@
+obs_studio:
+  port: 4444                  # The websocket port set in OBS
+  password: password          # The websocket password set in OBS
+  source_name: 'Media Source' # The name of the source which obs_video cues are played in
+  scene_name: Scene           # The name of the scene which the source is on
+  preload_time: 5             # The number of seconds before an obs_video track that the video is loaded and transitioned to
```
Comment (PeterJCLaw) [inline example-playlist.yaml]:
```suggestion
  preroll_time: 5             # The number of seconds before an obs_video track that the video is loaded and transitioned to
```

File: sr/comp/mixtape/cli.py [hunk 2]
```diff
@@ -81,6 +81,7 @@ def play(args):
             config['password'],
             config['source_name'],
             config['scene_name'],
+            config.get('preroll_time', 2),
```
Comment (PeterJCLaw) [inline sr/comp/mixtape/cli.py]: Could we make this non-optional?
```suggestion
            config['preroll_time'],
```

File: sr/comp/mixtape/mixtape.py [hunk 3]
```diff
@@ -130,6 +146,10 @@ def generate_play_actions(
             elif 'magicq_playback' in track:
                 action, name = self.get_run_cue_action(track, current_offset)
             elif 'obs_video' in track:
+                load_action, preroll_time = self.get_load_video_action(track, current_offset)
+                # priority 1 used since timing of load not critical
+                # and we don't want to delay other actions
+                yield track['start'] - preroll_time, 1, load_action
```
Comment (PeterJCLaw) [inline sr/comp/mixtape/mixtape.py:156]: Would be great to have an equivalent `print` for this scheduled item too.

File: sr/comp/mixtape/cli.py (hunk 2 above)
Comment (WillB97) [reply sr/comp/mixtape/cli.py]: Certainly, it was only optional for backward compatibility but given the feature is only 3 days old we don't really need to support that.

File: sr/comp/mixtape/mixtape.py (hunk 3 above)
Comment (WillB97) [reply sr/comp/mixtape/mixtape.py:156]: bd6430ae7872f7194a918a4cb763801e9b4faa5f

File: sr/comp/mixtape/cli.py (hunk 2 above)
Comment (PeterJCLaw) [reply sr/comp/mixtape/cli.py]: Yeah, I'm mixed on backwards compatibility at all for this sort of project. It's somewhat unlikely that anyone would be using this other than us and in any case it's very much a tool _for us_ and each use of it is pretty much a one-shot. I think the benefits of simpler code and (in this case) forcing it to be explicit outweigh the benefits of compatibility.

**Review by RealOrangeOne — APPROVED** [review]
Tested working

**Pull request merged**